cloud_sql_proxy/test_cloud_sql.py

```diff
@@ -1,11 +1,15 @@
 import json
```

Comment:
It it intentional rename for `cloud-sql-proxy` to `cloud_sql_proxy`? If so, it will be a braking change.

Reply:
Bazel can't reference paths containing hyphens ("-") so the renaming is necessary. What changes must be made to make this change a non-breaking one? Also, with the changes we're making here, the GCB presubmit will break because "python3 -m unittest" will fail because when DataprocTestCase tries importing absl, it will fail.

Reply:
I see, to make this change backward compatible we will need to change our deployment script to overwrite path when publishing to GCS.

But it seems that there are a [workaround](https://github.com/bazelbuild/bazel/issues/6841), may you try to use it instead of renaming?

Reply:
Also, we need to fix presubmit to use bazel, so we can test if this PR passes presubmit.

Reply:
The problem is that whatever rule builds our test will have to mentioned the hyphened path that leads to it. The only workaround for the hyphen issue that occurs to me is:
- leaving only the init action script in its original folder
- moving the tests to another unhyphened folder from which we can run tests referencing our script

For example, having:

- cloud-sql-proxy/cloud-sql-proxy.sh
- cloud-sql-proxy/BUILD  # just needed to export the script

- cloud_sql_proxy/test_cloud_sql.py  # references the init action for testing
- cloud_sql_proxy/pyspark_metastore_test.py
- cloud_sql_proxy/BUILD

Thoughts?

Reply:
I also tried this but it's not working for me: https://github.com/bazelbuild/bazel/issues/9171#issuecomment-532683449

Reply:
What about not having BUILD file in `cloud-sql-proxy` directory and configuring test target in the top-level BUILD file for it?

Reply:
I tried that but it still doesn't work. When I try referencing, for example, "cloud-sql-proxy/test_cloud_sql.py" as a srcs, it complains:

`Label '//:cloud-sql-proxy/test_cloud_sql.py' is invalid because 'cloud-sql-proxy' is a subpackage; perhaps you meant to put the colon here: '//cloud-sql-proxy:test_cloud_sql.py'?`

It expects a rule. So instead, I referenced the srcs as "//cloud-sql-proxy:test_cloud_sql.py" and exported the test Python file by declaring exports_files(["test_cloud_sql.py"]) making it accessible. This output another error:

`/usr/local/google/home/estebanlau/Documents/InitActionsFork/dataproc-initialization-actions/BUILD:15:9: in srcs attribute of py_library rule //:pyspark_metastore_test: //cloud-sql-proxy:pyspark_metastore_test.py: paths to Python packages may not contain '-' `

Reply:
Got it to work :) Apparently commenting out all the lines of the BUILD file in the cloud-sql-proxy folder is not sufficient. To be able to use cloud-sql-proxy files (e.g. test_cloud_sql) as a srcs, there can be no existing BUILD file so I deleted it.

Reply:
Cool, thanks for experimenting with this :)